scipy/spatial/transform/_rotation.pyx

```diff
@@ -223,6 +236,140 @@ cdef double[:, :] _compute_euler_from_matrix(
 
     return angles
 
+@cython.boundscheck(False)
+@cython.wraparound(False)
+cdef double[:, :] _compute_euler_from_quat(
+    np.ndarray[double, ndim=2] quat, const uchar[:] seq, bint extrinsic=False
+):
+    # The algorithm assumes extrinsic frame transformations. The algorithm
+    # in the paper is formulated for rotation quaternions, which are stored
+    # directly by Rotation.
+    # Adapt the algorithm for our case by
+    # 1. Replacing "d" with "-d" for intrinsic rotation
+    # 2. Reversing both axis sequence and angles for extrinsic rotations
+    
+    if not extrinsic:
+        seq = seq[::-1]
+
+    cdef int i = _elementary_basis_index(seq[0])
+    cdef int j = _elementary_basis_index(seq[1])
+    cdef int k = _elementary_basis_index(seq[2])
+    
+    # quick renormalization of transformed quaternion, assumes |q| = 1
+    cdef int norm_squared 
+
+    is_proper = i == k
+    if is_proper:
+        k = 3 - i - j # get third axis
+        norm_squared = 1
+    else:
+        norm_squared = 2
+        
+    # Step 0
+    # Check if permutation is even (+1) or odd (-1)     
+    cdef int sign = (i-j)*(j-k)*(k-i)//2
+
+    cdef Py_ssize_t num_rotations = quat.shape[0]
+
+    # some forward definitions
+    cdef double[:, :] angles = _empty2(num_rotations, 3)
+    cdef double[:] _angles # accessor for each rotation
```

Comment:
`_angles` is a bit awkward to read.

What if we do renames like this `angles -> angles_all or angles_result` and `_angles -> angles or angles_i (like the results for i-th quaternion)`.

Reply:
Personally I'm okay with both, I just decided to make the code as close to the old `_compute_euler_from_matrix` as possible.

Do you think changing that will create some kind of inconsistencies with the rest of the code?

Reply:
No, I don't think so. I'd say names likes `_angles` are not conventional.

It's not necessary to change it, do as you prefer.